Fix stopping dev server on Windows

feedmeapples opened the pull request:
<!--- Note to EXTERNAL Contributors -->
<!-- Thanks for opening a PR! 
If it is a significant code change, please **make sure there is an open issue** for this. 
We work best with you when we have accepted the idea first before you code. -->

<!--- For ALL Contributors 👇 -->

## What was changed
<!-- Describe what has changed in this PR -->

Starting killing the dev server process on Windows since sending signals (and SIGTERM or os.Interrupt) is not supported 

## Why?
<!-- Tell your future self why have you made these changes -->

Ensuring the dev server is stopped and doesn't leave hanging

## Checklist
<!--- add/delete as needed --->

1. Closes <!-- add issue number here -->

2. How was this tested:
<!--- Please describe how you tested your changes/how we can test them -->

Added test

3. Any docs updates needed?
<!--- update README if applicable
      or point out where to update docs.temporal.io -->

bergundy reviewed (commented):
See how to interrupt a process in windows:

https://github.com/temporalio/omes/blob/main/cmd/util_windows.go

feedmeapples commented:
> See how to interrupt a process in windows:
> 
> https://github.com/temporalio/omes/blob/main/cmd/util_windows.go

added here! Also not sure if it is actually meant to be interrupt in the `omes` repo. I've replaced this with `SIGTERM` in dev server

feedmeapples commented:
having unrelated issue with sdk-go v1.23.0. Recreated the PR on top of 1.21.2 until fixing CLI's CI with newer versions

feedmeapples commented on testsuite/devserver.go:
```diff
@@ -82,7 +81,7 @@ type DevServerOptions struct {
 // Temporal CLI based DevServer
 type DevServer struct {
-	cmd    *exec.Cmd
+	Cmd    *exec.Cmd
```
not a fan of making it public, though this seems to be the only option to test the RC after interrupting the server (especially for Windows)

feedmeapples replied:
unless we don't test this and only rely on `/cli`'s tests

cretz commented:
> having unrelated issue with sdk-go v1.23.0

Make sure the `go.temporal.io/api` and `go.temporal.io/sdk` are _both_ on the exact same versions as the `go.mod` of the server version you are using. Even if it compiles, there can be other issues if you mismatch. Server version should decide the other two.

cretz commented on testsuite/interrupt_windows.go:
```diff
@@ -0,0 +1,67 @@
+//go:build windows
```
This is not needed, it's assumed by the `_windows.go` filename

feedmeapples commented on testsuite/interrupt_windows.go:
```diff
@@ -0,0 +1,67 @@
+)
+
+// sendTerminate calls the break event on the given process for graceful shutdown.
+func sendTerminate(process *os.Process) error {
```
> > having unrelated issue with sdk-go v1.23.0
> 
> Make sure the `go.temporal.io/api` and `go.temporal.io/sdk` are _both_ on the exact same versions as the `go.mod` of the server version you are using. Even if it compiles, there can be other issues if you mismatch. Server version should decide the other two.

changed back to the `master` branch

cretz reviewed (approved):
Unfortunately we aren't running CI in Windows here like other ones, but LGTM and I am just hoping it works
